packages/shared/src/telemetry/collector.ts

```diff
@@ -110,7 +114,7 @@ export class TelemetryCollector {
 
     this.#logEvent(preparedPayload.event, preparedPayload);
 
-    if (!this.#shouldRecord(event.eventSamplingRate)) {
+    if (!this.#shouldRecord(event)) {
```

Comment:
We might want to pass `preparedPayload` here so we can include it as part of the throttled key, as if any of the "global" metadata that is provided in that method changes the event would still get throttled.

Reply:
Yeah, it makes sense! Better rely on the `preparedPayload` than the raw one.